src/util.cpp

```diff
@@ -537,6 +537,22 @@ class ArgsManagerHelper {
 
         return found_result;
     }
+
+    /* Special test for -testnet and -regtest args, because we
+     * don't want to be confused by craziness like "[regtest] testnet=1"
+     */
+    static inline bool GetNetBoolArg(const ArgsManager &am, const std::string& net_arg)
+    {
+        std::pair<bool,std::string> found_result(false,std::string());
+        found_result = GetArgHelper(am.m_override_args, net_arg, true);
+        if (!found_result.first) {
+            found_result = GetArgHelper(am.m_config_args, net_arg, true);
+            if (!found_result.first) {
+                return false; // not set
+            }
+        }
+        return InterpretBool(found_result.second); // is set, so evaluate
```

Comment:
It seems like code above this line is duplicating code in `ArgsManagerHelper::GetArg` in slightly inconsistent ways (passing getLast `[true, true]` to the two `GetArgHelper` calls instead of `[true, false]` and failing to acquire `cs_args`). Is there a reason for these inconsistencies and for duplicating code here instead of calling getArgs?

Reply:
Using `ArgsManagerHelper::GetArg` here would mean that calling `GetChainName()` a second (or third or..) time after `SelectParams(GetChainName())` could return a different result than the name of the chain actually being used if people do ridiculous things with their bitcoin.conf.

This should get cleaned up with some of the other config reworking that's going on though.

Reply:
I think `GetArgHelper(m_config_args, net_arg, true)` probably should have been `, false)`, well spotted.